src/provider/qti.js

```diff
@@ -492,18 +492,40 @@ var qtiProvider = {
             })
             .on('pause', function(data) {
                 const testContext = self.getTestContext();
-
+                let baseParams = {
+                    itemDefinition: testContext.itemIdentifier,
+                    reason: {
+                        reasons: data && data.reasons,
+                        comment: data && (data.originalMessage || data.message)
+                    }
+                }
                 this.setState('closedOrSuspended', true);
 
-                this.getProxy()
-                    .callTestAction('pause', {
-                        itemDefinition: testContext.itemIdentifier,
-                        itemState: self.itemRunner.getState(),
-                        reason: {
-                            reasons: data && data.reasons,
-                            comment: data && (data.originalMessage || data.message)
+                // get item response depends on availability item runner
+                const itemState = new Promise((resolve, reject) => {
+                    if(self.itemRunner.isCleared()){
+                        const proxy = this.getProxy();
+                        const store = proxy && proxy.itemStore;
+
+                        if(store) {
+                            store.get(testContext.itemIdentifier)
+                            .then(result => {
+                                resolve(result.itemState);
+                            })
+                            .catch(err => reject(err));
+                        } else {
+                            reject(__('Response can not be stored.'));
                         }
-                    })
+                    } else {
+                        resolve(self.itemRunner.getState());
+                    }
+                });
+
+                itemState.then(data =>{
+                    const params = Object.assign({}, baseParams, {itemState: data});
```

Comment:
I have doubts about this approach in general (sorry, don't have enough knowledge about this):

- Instead of getting state from `proxy.itemStore` (are there any guarantees on what and in which state is inside?), could we pass `itemState: null` to `pause` request and ensure backend will keep value already submitted to server in this case?

- it feels strange to rely on itemRunner (`itemRunner.isCleared()`) to detect this, feels like it should be some test-runner state instead, but I don't see anything like test-runner state here.... https://github.com/oat-sa/tao-test-runner-qti-fe/blob/develop/src/provider/qti.js#L234 `context.itemSessionState` but what value does it have here?

- Is it guaranteed that `pause` action (initiated by system) will happen _only immediately after_ `move` action (initiated by user). Because if not, and it can happen in the middle of user answering, we will probably submit answer he didn't finish - is it fine? Shouldn't unfinished answer be saved only if there's `answerCachePlugin`? Though I see passing`itemState` on `pause` was in code before (`itemState: self.itemRunner.getState()` on line 501)

Reply:
Totally agree Olga. Very good catch and analysis.

Despite the proposed solution looks to fix the issue, it does not suit well. Actually, it only takes care of one case, which is only occurring when we have both the proctoring and the pre-caching activated.

Moreover, it assumes too much on the proxy internals, which is not safe.

The pause event can occur at any time. The state should be supplied when pausing the test to prevent the user from losing any data. For this particular point, I don't think it is an issue if the user input is not complete. You can consider it as a backup from an interruption. When resuming, the user expects to recover from the left state.

In fact, the solution needs to take care of all cases, but without dealing with consumed components internals. The test needs to maintains and observe the states, and then react accordingly, but without looking up inside the guts of the components.